pennylane_lightning/src/Gates.cpp

```diff
@@ -70,6 +70,14 @@ void Pennylane::AbstractGate::applyKernel(const StateVector& state, const std::v
     }
 }
 
+const double Pennylane::AbstractGate::generatorScalingFactor{};
+void Pennylane::AbstractGate::applyGenerator(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices) {
+    (void)state;
+    (void)indices;
+    (void)externalIndices;
+    throw NotImplementedException();
+}
```

Comment:
Would eventually other gates define an `applyGenerator` function or just the ones where there's a definition provided at the moment?

Reply:
Perhaps eventually, if we figure out a way to define the generators for those gates. Although to be honest I'm not sure how that'd be done, or if we'd need to solve it in some other way.